ISISNeutronMuon/MDANSE
# Add MDAnalysis converter

**Description of work**
Added MDAnalysis converter so that a larger range of files can now be converted to the MDANSE format. Trajectories are now loaded up with a topology file and optionally a multiple number of coordinate files, see https://userguide.mdanalysis.org/stable/formats/index.html#formats for details. In MDAnalysis atom can be group following this scheme

![image](https://github.com/user-attachments/assets/5c415345-2801-42b9-89fd-ab8c0dd9c69f)

I have not mapped the MDAnalysis atom grouping into the MDANSE atom grouping system as it is not clear what exactly the mapping should be and whether we are keeping the chemical entity objects in the short/medium term. 

Features and changes
- The input file format for the topology and coordinate format can be specified or the AUTO setting can be used to let MDAnalysis determine the format.
- Multiple files can be loaded up into the coordinates files. MDAnalysis will stich together these trajectories. A continous frame stiching option has been added, see https://userguide.mdanalysis.org/stable/reading_and_writing.html.
- Atom mapping has be updated so that the selection boxes are in a scrollable area. This should be better for the larger system that can now be converted.

Closes #585 

**To test**
1) Convert some gromacs trajectories from the MDANSE-Examples repository. 
Enter PDB files into the topology setting and trr or xtc files should be entered into coordinate files setting. Once entered the atom mapping and time step should update automatically. Check that the files convert with any issues
2) Convert a DL POLY trajectory.
Convert the dlpoly trajectory using the history file from MDANSE-Examples. For this case we only need to enter a file in the topology file input. As the file extension is not added to these files you will need to change the combobox to HISTORY. Once converted check that the files convert with any issues.
3) Try converting a trajectory using multiple coordinate files.
Either generate multiple coordinate files or copy and rename them. Convert the trajectory, if you copy and renamed it then the MDANSE trajectory should be twice as long and repeat itself.
4) Try converting some trajectories from https://www.ccpbiosim.ac.uk/slim-md/slimmd-database, most of these should convert without any issues.
